### PR TITLE
Pass system environment variables when executing custom command

### PR DESCRIPTION
**Pull request by woohyunjin06**
### Short description 📝

> This PR allows sub process (e.g. Plugin) to have access to system environment variables

### How to test the changes locally 🧐

> Run custom command that require environment variables, run unit test

### Checklist ✅

- [X] The code architecture and patterns are consistent with the rest of the codebase.
- [X] The changes have been tested following the [documented guidelines](https://docs.tuist.io/contributors/testing-strategy/).
- [ ] The PR includes the label `changelog:added`, `changelog:fixed`, or `changelog:changed`, whenever it should be included in the “Added”, “Fixed” or “Changed” section of the CHANGELOG. Note: when included in the CHANGELOG, the title of the PR will be used as entry, please make sure it is clear and suitable.
- [ ] In case the PR introduces changes that affect users, the documentation has been updated.
- [ ] In case the PR introduces changes that affect how the cache artifact is generated starting from the `TuistGraph.Target`, the `Constants.cacheVersion` has been updated.

### CODE REVIEW

File: Sources/TuistSupportTesting/Utils/MockSystem.swift [hunk 1]
```diff
@@ -9,12 +9,21 @@ public final class MockSystem: Systeming {
 
     // swiftlint:disable:next large_tuple
     public var stubs: [String: (stderror: String?, stdout: String?, exitstatus: Int?)] = [:]
+    public var requiredEnvironmentVariables: [String: [String: String]] = [:]
```
[the remark this answers is not in the document]
Comment (fortmarek) [reply Sources/TuistSupportTesting/Utils/MockSystem.swift]: This is great! Please, create a PR against the tuist's `ExampleTuistPlugin` repo and I'll do a review there 👍

File: projects/tuist/fixtures/app_with_plugins/Tuist/Config.swift [hunk 2]
```diff
@@ -4,6 +4,6 @@ let config = Config(
     plugins: [
         .local(path: .relativeToManifest("../../LocalPlugin")),
         // TODO: Change back to tuist organization
-        .git(url: "https://github.com/fortmarek/ExampleTuistPlugin", tag: "0.4.0"),
+        .git(url: "https://github.com/woohyunjin06/ExampleTuistPlugin", tag: "3.1.0"),
```
Comment (danyf90) [inline projects/tuist/fixtures/app_with_plugins/Tuist/Config.swift:7]: Can't we just do the change in `https://github.com/tuist/ExampleTuistPlugin`?

Also, why did you need to change the executable below?

**Review by danyf90 — CHANGES_REQUESTED** [review]

File: projects/tuist/fixtures/app_with_plugins/Tuist/Config.swift (hunk 2 above)
Comment (woohyunjin06) [reply projects/tuist/fixtures/app_with_plugins/Tuist/Config.swift:6]: I've confirmed that this url does not affect the new test case, and I have updated the repository url.

The reason for change the executable is that test case I created in this PR use the updated `create-file` command to check whether environment variable is defined or not.

**Comment by danyf90** [comment]
@woohyunjin06 I have updated the PR to use the new plugin, could you please have a look that everything is ok?

**Review by danyf90 — APPROVED** [review]

**Comment by woohyunjin06** [comment]
Thanks @danyf90

The test scenario currently written cannot be completed successfully without `tuist-create-file` executable file.
So I changed scenario to  execute `create-file` task in the project directory. 

I don't know the exact intension of executing `create-file` task in the current directory, so could you check if it is ok?
